lighthouse-core/report/v2/renderer/details-renderer.js

```diff
@@ -134,3 +164,14 @@ DetailsRenderer.ListDetailsJSON; // eslint-disable-line no-unused-expressions
  * }}
  */
 DetailsRenderer.CardsDetailsJSON; // eslint-disable-line no-unused-expressions
+
+/**
+ * @typedef {{
+ *     type: string,
+ *     text: string,
+ *     url: (string|undefined),
+ *     source: string,
```

Comment:
`url` is where the code came from :) I stuck to `source` b/c it's what the protocol uses: `logEntry.source`.